src/widgets/lv_img.h

```diff
@@ -87,14 +108,61 @@ lv_obj_t * lv_img_create(lv_obj_t * parent);
  *====================*/
 
 /**
- * Set the image data to display on the object
+ * Some image format support animation. In that case, you might want to control it.
+ * @param obj       pointer to an image object
+ * @param ctrl      A ORed mask of lv_img_ctrl_t flags you want
+ */
+void lv_img_set_play_mode(lv_obj_t * obj, const lv_img_ctrl_t ctrl);
+
+/**
+ * Set the current frame of the animation
+ * @param obj       pointer to an image object
+ * @param index     Index of the frame to reach
+ * @warning This will only work once the source of the image is set and the image is animated
+ */
+lv_res_t lv_img_set_current_frame(lv_obj_t * obj, const lv_frame_index_t index);
+
+/**
+ * Set the stop frame for the animation
  * @param obj       pointer to an image object
- * @param src_img   1) pointer to an ::lv_img_dsc_t descriptor (converted by LVGL's image converter) (e.g. &my_img) or
+ * @param index     Index of the frame to stop at. Will emit a EVENT_READY when reached.
+ * @param forward   If 1, the animation plays forward, else plays backward
+ * @warning This will only work once the source of the image is set and the image is animated
+ */
+lv_res_t lv_img_set_stopat_frame(lv_obj_t * obj, const lv_frame_index_t index, const int forward);
+
+/**
+ * Parse the image source to display on the the object
+ *
+ * @param obj       pointer to an image object
+ * @param src       1) pointer to an ::lv_img_dsc_t descriptor (converted by LVGL's image converter) (e.g. &my_img) or
  *                  2) path to an image file (e.g. "S:/dir/img.bin")or
  *                  3) a SYMBOL (e.g. LV_SYMBOL_OK)
+ *                  4) pointer to a lv_img_src_t instance
+ *
+ * @deprecated This function is deprecated in favor of lv_img_accept_src
+ *             You can create a lv_img_src_t * from a file, symbol or plain data via lv_img_src_set_file/symbol/data
+ *
  */
 void lv_img_set_src(lv_obj_t * obj, const void * src);
 
+
+/** Set the source of this image to a lv_img_src_t and check if it's usable.
+ *
+ * Some picture don't have intrinsic size (like a vector based format, SVG, Lottie), or contain multiple
+ * parts (like multiple frames for animated format, GIF, Lottie).
+ * If you need to deal with such format, you might have to call other function
+ * (like `lv_obj_set_size` for vector format) before calling this
+ *
+ * @param obj           pointer to an image object
+ * @param src           pointer to an image source object
+ * @param skip_cache    if 1, the image will only be parsed to extract its metadata but it'll skip
+ *                      the image cache until it's actually used.
+ *                      You might only need to set this if you are preparing your screen at once on start
+ * @return LV_RES_OK    If the image was correctly parsed
+ */
+lv_res_t lv_img_accept_src(lv_obj_t * obj, const lv_img_src_t * src, int skip_cache);
```

Comment:
I like the return value here but I'm not sure about `skip_cache`. I think we should be able to handle it internally.  It'd be great to know what we win by having this option. E.g. what is the different in widget creation time when creating 10 images with our without caching. 

I'd like to avoid solving a non existing (or only theoretical) problem.

---

```suggestion
lv_res_t lv_img_set_src_ext(lv_obj_t * obj, const lv_img_src_t * src, int skip_cache);
```

The word `accept` is not that intuitive for me.

Reply:
That's a requirement you've made in the former PR. It was made for those GUI where everything is set up/declared upfront (like when creating all screens). We can remove this here and use a caps instead (so in case you want to gain 0.1% of speed compared to hitting the cache, you'll set the decoder's context cap to "SKIP_INITIAL_CACHE" or something), just like we do for lottie files and their initial size.  

Ok for accept/set_src_ex.

My initial idea was to have a different name than `set_src` to avoid LVGL <= 8 code to compile when we drop the `void*` source type. In LVGL 9, it can't be `lv_img_set_src` (else previous code will compile but will crash at runtime). `set_src_ex` without a `set_src` looks awkward too. I don't know.

Reply:
> so in case you want to gain 0.1% of speed 

It seems much more than 0.1%. E.g. decoding a PNG image vs. reading its first few bytes to get its resolution.

This issue emerged due to the removal of `decoder->info_cb`, right? Are you sure it's really needs to be removed? Just thinking aloud: if getting the info, and decoding takes the same time the decoder can decide to save the decoded image too.

Reply:
The `decoder_accept` doesn't decode the image (unlike get_info), only the headers. That's true for simple format (like bmp, gif, png, jpeg) but not for ffmpeg (as I haven't found a way to do that) or rlottie.
In this latter case, I'm still skipping rendering/color conversion, so it still faster than full image decoding.

By default, skip_cache is 0, so instead of calling decoder_accept, we directly call decoder_open that's decoding the image (or part of the image) and store the result in the cache. So no time is wasted if the next action will be to display the image.

When skip_cache is 1, `decoder_accept` is called instead. For simple format, it avoid the cache and is faster than decoding the whole picture. When painting, since the cache is empty, decoder_open will be called. 
For more complex format, the image will be parsed and decoded but image rendering and video decoding will be skipped.
However as soon as the image need painting, we'll have to pay the parsing time again.
 
In current LVGL code, get_info decodes the image and throw the result away. Worst, it's called twice per paint (just before a decoder_open) so it's a lot slower. It's only faster when dealing with SRC_VARIABLE, because it's using the `lv_img_dsc_t`'s fields, but it's like cheating, since another process gave it the exams' results.

In the end, I'm not sure of the utility of skip_cache. If you have plenty of screens you're building at once on program startup, without skip_cache, they'll all compete on the cache, so we might get a lot of evictions, but at the same time, the last screen will have its images cached already. Maybe it's worth adding a caps to the decoder and have the generic screen code API (upcoming for v8.3 ???) set this caps internally. This would gives a good advantage but doesn't modify the API with some useless complexity, IMHO.

Reply:
I see `skip_cache` is unused in the current PR. TBH, I can't exactly oversee the consequences blindly (i.e. without seeing the actual processes).

I suggest removing the `skip_cache` parameters for now, and reconsider it when the functions using it will be added too.